### DISCUSSION TITLE
Question Relating to Starting a Blog Post and Customization

### DISCUSSION THREAD

**Discussion by rafay99-epic** [incluud/accessible-astro-starter#64] [Q&A]
# Intro:
Hi, Abdul Rafay! I'm here to help you with your questions regarding creating a blog post and understanding the codebase for your website. I'll do my best to assist you. Let's get started!

# Questions:
1. How can I create a blog post if I am using this theme?
2. I also want to understand the codebase so that I can modify it to fit my needs. Can you explain the folder structure or the inner workings of the codebase?

**Comment by markteekman** [comment]
Hey @rafay99-epic! Sorry for leaving this unanswered for a long time, must have slipped my attention.. 😓 I'm working on a documentation website where I'm hoping to explain this all so stay tuned!

**Reply by rafay99-epic** [reply]
No worries, I get it. Honestly, when I first asked this question, I didn’t really understand the Astro framework. But now, I’ve got a much better grasp of the structure. I’m still referring to the docs, though. Let me know if you ever need any help!

**Reply by p2635** [reply]
This post is a bit confusing, it starts off with "Hi, Abdul Rafay! I'm here to help you with your questions regarding creating a blog post and understanding the codebase for your website. I'll do my best to assist you. Let's get started!". But the whole post is actual the question. 

Anyway guys, I need help on the same thing. Can someone please reopen this discussion as a task to update the docs? I still don't understand how to create blogs as pages. 

@rafay99-epic @markteekman

**Reply by markteekman** [reply]
@p2635 I understand the confusion. I thought @rafay99-epic meant he was missing documentation. To be fair, we use [Astro's Content Collections](https://docs.astro.build/en/guides/content-collections/), and they have excellent documentation. The portfolio in the starter is built the same way it's explained here. The blog in the starter is an example of getting external blog items from another source. The easiest thing you could do is remove all blog pages and rename portfolio to blog 🙂 If you're using AI, you can also use our [Astro Agent Skills](https://github.com/incluud/astro-agent-skills) to help you out.

Let me know if that works for you!

**Reply by rafay99-epic** [reply]
@p2635 I completely understand the confusion. When I first started building my content/blog website with Astro, I honestly had no idea where to begin.

I started by looking at different templates and themes, but at the time, I didn't fully understand how Astro worked—how the folder structure was organized, how pages were rendered, how Content Collections worked, or how everything fit together.

Eventually, after spending a lot of time reading the Astro documentation, experimenting with different approaches, and watching tutorials, things started to click. By the time Mark answered some of my questions and mentioned that he was working on the documentation, I had already gone quite far down that path.

At that point, I decided to build my own Astro blog, components, and theme system from scratch. I wanted something fully customized to my needs rather than relying on a generic theme or template.

If you're currently trying to understand this theme or Astro's overall structure, I'd highly recommend **Coding in Public** on YouTube. His Astro videos helped me a lot when I was getting started, especially with understanding project structure, components, and how Astro works in general.

I'd also strongly recommend going through the official Astro documentation alongside the videos. This is particularly important because Astro has changed quite a bit across versions 4, 5, 6, and now 7, so some older tutorials and implementations may not exactly match what you're working with today.

And if you have any specific questions about the codebase currently implemented by @markteekman and his team, I'm more than happy to help wherever I can.

**Reply by rafay99-epic** [reply]
@p2635, if you need more code examples or Astro components, you can also check out my website's codebase: github.com/rafay99-epic/rafay99.com

This isn't a promotion. Explore different themes, see how they work, and take whatever fits your needs. Trust your own process and instincts as you build. You'll understand Astro much better by experimenting with different approaches and figuring out what works for you.

@markteekman, thank you for creating this project. Without it, I probably wouldn't have understood the Astro framework and the inner workings of an Astro site as well as I do today. I especially love the SEO side of the project and learned a lot from it.

Again, thank you so much for building and sharing it!
